stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal

```diff
@@ -0,0 +1,234 @@
+    public boolean httpOnly = false;
+    public boolean secure = false;
+    public time:Time creationTime = time:currentTime();
+    public time:Time lastAccessedTime = time:currentTime();
+    public boolean hostOnly = false;
+
+    // Returns false if the cookie will be discarded at the end of the "session"; true otherwise.
+    public function isPersistent() returns boolean {
+        if (self.expires == "" && self.maxAge == 0) {
+            return false;
+        }
+        return true;
+    }
+
+    // Returns true if the attributes of the cookie are in the correct format; false otherwise.
+    public function isValid() returns boolean | error {
+        error invalidCookieError;
+        if (self.name == "" || self.value == "") {
+            invalidCookieError = error("Empty name value pair");
+            return invalidCookieError;
+        }
+        if (self.domain != "") {
+            self.domain = self.domain.toLowerAscii();
+        }
+        if (self.domain.startsWith(".")) {
+            self.domain = self.domain.substring(1, self.domain.length());
+        }
+        if (self.domain.endsWith(".")) {
+            self.domain = self.domain.substring(0, self.domain.length() - 1);
+        }
+        if (self.path != "" && (!self.path.startsWith("/") || stringutils:contains(self.path, "?"))) {
+            invalidCookieError = error("Path is not in correct format ");
+            return invalidCookieError;
+        }
+        if (self.expires != "" && !toGmtFormat(self)) {
+            invalidCookieError = error("Time is not in correct format");
+            return invalidCookieError;
+        }
+        if (self.maxAge < 0) {
+            invalidCookieError = error("Max Age is less than zero");
+            return invalidCookieError;
+        }
+        return true;
+    }
+
+    // Gets the Cookie object in its string representation to be used in the ‘Set-Cookie’ header of the response.
+    function toStringValue() returns string {
+        string setCookieHeaderValue = "";
+        setCookieHeaderValue = appendNameValuePair(setCookieHeaderValue, self.name, self.value);
+        if (self.domain != "") {
+            setCookieHeaderValue = appendNameValuePair(setCookieHeaderValue, "Domain", self.domain);
+        }
+        if (self.path != "") {
+            setCookieHeaderValue = appendNameValuePair(setCookieHeaderValue, "Path", self.path);
+        }
+        if (self.expires != "") {
+            setCookieHeaderValue = appendNameValuePair(setCookieHeaderValue, "Expires", self.expires);
+        }
+        if (self.maxAge > 0) {
+            setCookieHeaderValue = appendNameIntValuePair(setCookieHeaderValue, "Max-Age", self.maxAge);
+        }
+        if (self.httpOnly == true) {
+            setCookieHeaderValue = appendOnlyName(setCookieHeaderValue, "HttpOnly");
+        }
+        if (self.secure == true) {
+            setCookieHeaderValue = appendOnlyName(setCookieHeaderValue, "Secure");
```

Comment:
Shouldn't we use constants for these? "HttpOnly", "Secure", etc.?

Reply:
So do we need to use constants for "domain", "path", "expires"... also??

Reply:
+1 from me.